salt/_utils/kubernetes_utils.py

```diff
@@ -0,0 +1,535 @@
+        group, _, version = api_version.rpartition('/')
+        if not (group and version):
+            raise ValueError(
+                "Malformed 'apiVersion': {} "
+                "(expected format '<group>/<version>')".format(api_version)
+            )
+
+        self._group = group
+        self._version = version
+        self._kind = kind
+        scope = ObjectScope(scope)
+        self._scope = scope
+        self._plural = plural
+
+        self._create = getattr(CustomObjectsApi, self._method('create'))
+        self._retrieve = getattr(CustomObjectsApi, self._method('get'))
+        self._update = getattr(CustomObjectsApi, self._method('patch'))
+        self._delete = getattr(CustomObjectsApi, self._method('delete'))
+        self._replace = getattr(CustomObjectsApi, self._method('replace'))
+
+    group = property(operator.attrgetter('_group'))
+    version = property(operator.attrgetter('_version'))
+    kind = property(operator.attrgetter('_kind'))
+    scope = property(operator.attrgetter('_scope'))
+    plural = property(operator.attrgetter('_plural'))
+    create = property(operator.attrgetter('_create'))
+    retrieve = property(operator.attrgetter('_retrieve'))
+    update = property(operator.attrgetter('_update'))
+    delete = property(operator.attrgetter('_delete'))
+    replace = property(operator.attrgetter('_replace'))
+
+    @property
+    def api_version(self):
+        return '/'.join([self.group, self.version])
+
+    @property
+    def key(self):
+        # Used for indexing in `KNOWN_CUSTOM_OBJECTS`
+        return (self.api_version, self.kind)
+
+    def _method(self, verb):
+        return '{}_{}_custom_object'.format(verb, self.scope.value)
+
+
+_CUSTOM_OBJECTS = [
+    CRDetails('monitoring.coreos.com/v1', 'Alertmanager',
+              scope='namespaced', plural='alertmanagers'),
+    CRDetails('monitoring.coreos.com/v1', 'Prometheus',
+              scope='namespaced', plural='prometheuses'),
+    CRDetails('monitoring.coreos.com/v1', 'PrometheusRule',
+              scope='namespaced', plural='prometheusrules'),
+    CRDetails('monitoring.coreos.com/v1', 'ServiceMonitor',
+              scope='namespaced', plural='servicemonitors'),
+]
+
+KNOWN_CUSTOM_OBJECTS = {details.key: details for details in _CUSTOM_OBJECTS}
+
+
+class _DictWrapper(object):
+    """Wrapper for dynamic attribute access support over a simple dict.
+
+    Used by `CustomObject` to emulate the other models for `kubernetes.client`.
+    """
+    def __init__(self, fields):
+        self._fields = {
+            key: self.from_value(value) for key, value in fields.items()
+        }
+
+    @classmethod
+    def from_value(cls, value):
+        if isinstance(value, dict):
+            return cls(value)
+        if isinstance(value, list):
+            return [cls.from_value(val) for val in value]
+        return value
+
+    def __repr__(self):
+        return repr(self._fields)
+
+    def __getattribute__(self, name):
+        try:
```

Comment:
Why not:

```python
        try:
            return super(_DictWrapper, self).__getattribute__(name)
        except AttributeError:
            try:
                return self._fields[name]
            except KeyError:
                raise AttributeError(
                    "Custom object has no attribute '{}'".format(name)
                )
```
Then you avoid the `pass` and the need for a comment.

Reply:
Sure, will do.